### PR TITLE
HOTFIX: fix compile errors

### PR DESCRIPTION
**Pull request by mjsax**
Some internal APIs got removed via
https://github.com/apache/kafka/pull/10872

### CODE REVIEW

**Comment by mjsax** [comment]
Cf https://github.com/confluentinc/ksql/pull/7694

**Review by vcrfxia — APPROVED** [review]
Thank you @mjsax !! You are a hero 🦸 ❤️

File: src/main/java/io/confluent/examples/streams/microservices/OrderDetailsService.java [hunk 1]
```diff
@@ -101,7 +97,7 @@ private void startService(final String bootstrapServers, final Properties defaul
             }
           }
           if (eosEnabled) {
-            producer.sendOffsetsToTransaction(consumedOffsets, CONSUMER_GROUP_ID);
+            producer.sendOffsetsToTransaction(consumedOffsets, new ConsumerGroupMetadata(CONSUMER_GROUP_ID));
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/microservices/OrderDetailsService.java:100]: Fixing deprecation warning on the side

File: src/main/scala/io/confluent/examples/streams/algebird/CMSStoreChangeLogger.scala [hunk 2]
```diff
@@ -37,7 +37,7 @@ class CMSStoreChangeLogger[K, V](val storeName: String,
   private val collector = context.asInstanceOf[RecordCollector.Supplier].recordCollector
 
   def this(storeName: String, context: StateStoreContext, serialization: StateSerdes[K, V]) = {
-    this(storeName, context, context.taskId.partition, serialization)
+    this(storeName, context, context.taskId.partition(), serialization)
```
Comment (mjsax) [inline src/main/scala/io/confluent/examples/streams/algebird/CMSStoreChangeLogger.scala:40]: Fixing deprecation warning on the side

File: src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java [hunk 3]
```diff
@@ -45,7 +44,7 @@
 public class MicroserviceTestUtils {
 
   private static final Logger log = LoggerFactory.getLogger(MicroserviceTestUtils.class);
-  private static final List<TopicTailer> tailers = new ArrayList<>();
+  private static final List<TopicTailer<?, ?>> tailers = new ArrayList<>();
```
Comment (mjsax) [inline src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java:47]: Fixing warning on the side. (also below)

**Pull request merged**